sqlglot/dialects/clickhouse.py

```diff
@@ -328,3 +352,27 @@ def parameterizedagg_sql(self, expression: exp.Anonymous) -> str:
 
         def placeholder_sql(self, expression: exp.Placeholder) -> str:
             return f"{{{expression.name}: {self.sql(expression, 'kind')}}}"
+
+        def oncluster_sql(self, expression: exp.OnCluster) -> str:
+            return f"ON CLUSTER {self.sql(expression, 'this')}"
+
+        def createable_sql(
+            self,
+            expression: exp.Create,
+            locations: dict[exp.Properties.Location, list[exp.Property]],
+        ) -> str:
+            kind = self.sql(expression, "kind").upper()
+            if kind in self.ON_CLUSTER_TARGETS and locations.get(exp.Properties.Location.POST_NAME):
+                this_name = self.sql(expression.this, "this")
```

Comment:
was this really necessary? seems like a lot of copy and paste? can you just do this with flags in the generator?

Reply:
Handling `pretty=True` should better be uniform, I think.

Reply:
yea but can't you make this method more dry? teradata and clickhouse are copy pasted

Reply:
It's more DRY now, and also improved the pretty printer in Teradata